subset/switches/src/main/java/switchtest/SwitchInterrogator.java

```diff
@@ -218,99 +229,111 @@ public void receiveData(String data) {
 
-      if (requestFlag == (interfacePos + 1)) {
-        parse_single(
-            temp_data, show_interface_expected, show_interface_pointers, show_interface_data);
-      } else if (requestFlag == (platformPos + 1)) {
-        parse_single(temp_data, show_platform_expected, show_platform_pointers, show_platform_data);
+        if (requestFlag == (interfacePos + 1)) {
+          parse_single(
+              temp_data, show_interface_expected, show_interface_pointers, show_interface_data);
+        } else if (requestFlag == (platformPos + 1)) {
+          parse_single(
+              temp_data, show_platform_expected, show_platform_pointers, show_platform_data);
+        }
       }
+    } catch (Exception e) {
+      System.err.println("Exception parse_packet:" + e.getMessage());
     }
   }
 
   private void parse_single(
       String raw_data, String[] expected_array, int[] pointers_array, String[] data_array) {
-    int start = 0;
-    for (int x = 0; x < expected_array.length; x++) {
-      start = recursive_data(raw_data, expected_array[x], start);
-      if (start == -1) {
-        start = pointers_array[x - 1];
-        pointers_array[x] = -1;
-      } else {
-        pointers_array[x] = start;
+    try {
+      int start = 0;
+      for (int x = 0; x < expected_array.length; x++) {
+        start = recursive_data(raw_data, expected_array[x], start);
+        if (start == -1) {
+          start = pointers_array[x - 1];
+          pointers_array[x] = -1;
+        } else {
+          pointers_array[x] = start;
+        }
       }
-    }
 
-    for (int x = 0; x < data_array.length; x++) {
-      int chunk_start = x * 2;
-      int chunk_end = 1 + (x * 2);
+      for (int x = 0; x < data_array.length; x++) {
+        int chunk_start = x * 2;
+        int chunk_end = 1 + (x * 2);
 
-      int chunk_s = pointers_array[chunk_start];
-      int chunk_e = pointers_array[chunk_end];
+        int chunk_s = pointers_array[chunk_start];
+        int chunk_e = pointers_array[chunk_end];
 
-      if (chunk_s > 0) {
-        if (chunk_e == -1) chunk_e = raw_data.length();
+        if (chunk_s > 0) {
+          if (chunk_e == -1) chunk_e = raw_data.length();
 
-        data_array[x] = raw_data.substring(chunk_s, chunk_e);
+          data_array[x] = raw_data.substring(chunk_s, chunk_e);
 
-        data_array[x] = data_array[x].substring(expected_array[chunk_start].length());
+          data_array[x] = data_array[x].substring(expected_array[chunk_start].length());
 
-        String extracted_data = expected_array[chunk_start] + data_array[x];
+          String extracted_data = expected_array[chunk_start] + data_array[x];
 
-        if (debug) System.out.println(extracted_data);
+          if (debug) System.out.println(extracted_data);
 
-        login_report += extracted_data + "\n";
+          login_report += extracted_data + "\n";
+        }
       }
+    } catch (Exception e) {
+      System.err.println("Exception parse_single:" + e.getMessage());
```

Comment:
why are you just silently ignoring these?

Reply:
only useful when debugging the code during development, 
shouldnt happen in a real world